sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -53,7 +53,7 @@ namespace Azure { namespace Core { namespace Http {
   /**
    * @brief HTTP transport layer error.
    */
-  class TransportException : public Azure::Core::RequestFailedException {
+  class TransportException final : public Azure::Core::RequestFailedException {
```

Comment:
I don't think we should put final on exception types like these. User's should be able to derive from these to build up the exception heirarchy.

Reply:
I don't think we do at this moment view our exception classes being consumed by customers deriving from them.. Our exception is "an exception from Azure SDK", and user's code is not Azure SDK, so I don't see why they should inherit.

But even if this the case, we should be removing final as response to a demand. It is hard to guess what would be needed and what would not, this is why with this change we seal all the ones we are not sure about, leaving only the ones that are intended for inheritance (i.e. base policy), and that are being inherited by our SDK classes.